src/Gd/Imagine.php

```diff
@@ -87,9 +87,14 @@ public function open($path)
         $path = $loader->getPath();
 
         $data = $loader->getData();
-        $resource = $this->withoutExceptionHandlers(function () use (&$data) {
-            return @imagecreatefromstring($data);
-        });
+
+        if($this->isWebp($data)) {
```

Comment:
`imagecreatefromwebp` has been introduced in PHP 5.4, but Imagine [supports PHP 5.3.2](https://github.com/avalanche123/Imagine/blob/1.2.0/composer.json#L20). so what about

```suggestion
        if(function_exists('imagecreatefromwebp') && $this->isWebp($data)) {
```